scripts/xe-install-supplemental-pack

```diff
@@ -32,6 +32,6 @@ iso="$1"
 [ -n "$iso" ] || usage
 
 update_uuid=$(xe update-upload file-name="$iso" ${sr_uuid:+sr-uuid=$sr_uuid})
-trap "xe update-pool-clean uuid=$update_uuid" EXIT ERR
-trap "xe update-destroy uuid=$update_uuid" ERR
+trap "xe update-pool-clean uuid=$update_uuid; xe update-destroy uuid=$update_uuid" ERR
```

Comment:
If the `update-pool-clean` fails then the script will terminate without calling `update-destroy`. You can fix this by having `set +e` first.